### ISSUE TITLE
zipAll doesn't call complete on an empty stream

### ISSUE THREAD

**Issue by KMax** [ReactiveX/rxjs#2426]
**RxJS version:**
5.0.1 (locally)
I don't know which version is used in JS Bin _(link below)_.
**Code to reproduce:**
Source: http://jsbin.com/jeyuqojopi/edit?js,console

```
const source = Rx.Observable.empty();

source.map((v) => Rx.Observable.of(v+1)).subscribe({
  next: (_) => console.log("next"),
  complete: () => console.log("complete")
});

source.map((v) => Rx.Observable.of(v+1)).zipAll().subscribe({
  next: (v) => console.log(v),
  complete: () => console.log("complete")
});
```
**Expected behavior:**
In console:
```
"complete"
"complete"
```
**Actual behavior:**
In console:
```
"complete"
```
**Additional information:**
I'm not sure whether it's a bug. If it's not a bug, could you share an idea how to achieve the expected behavior. Thank you!

**Comment by KMax** [comment]
A workaround is to replace `.zipAll()` with `.combineAll()` method. Not sure what's the difference between them, but it works for my use case.

**Comment by martinsik** [comment]
@KMax Isn't the `zip()` example equivalent to the following?:

```
Rx.Observable.zip(source.map((v) => Rx.Observable.of(v+1))).subscribe({
  next: (v) => console.log(v),
  complete: () => console.log("complete")
});
```

This will send proper `complete` notification.

**Comment by KMax** [comment]
@martinsik It looks like an equivalent, yes. Thank you!

**Comment by kwonoj** [comment]
closed via #2451.
